src/bluecoins_cli/adb.py

```diff
@@ -0,0 +1,70 @@
+import datetime
+import subprocess
+
+from adbutils import adb  # type: ignore
+from adbutils._device import AdbDevice  # type: ignore
+
+
+class Device:
+    def __init__(self, device: AdbDevice) -> None:
+        self.device = device
+
+        self.APP_ID = 'com.rammigsoftware.bluecoins'
+
+        current_time = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
+        self.DB = f"bluecoins-{current_time}"
+
+    @classmethod
+    def connect(cls) -> 'Device':
+        device_list = adb.device_list()
+
+        if not device_list:
+            raise Exception('Device is not found')
+        elif len(device_list) >= 2:
+            raise Exception('Found two or more devices. Connect only one device.')
+
+        device = adb.device(serial=device_list[0].serial)
+        return Device(device)
+
+    def stop_app(self) -> None:
+        self.device.app_stop(self.APP_ID)
+        self.device.shell(f'pm disable-user --user 0 {self.APP_ID}')
+
+    def get_app_user_id(self) -> int:
+        user_response = self.device.shell(f'dumpsys package {self.APP_ID} | grep userId')
+        # split to list, take second element, str --> int
+        return int(user_response.split(sep='=')[1])
+
+    def pull_db(self) -> None:
+        subprocess.run(
+            f'adb shell su {self.APP_ID} -c "cat /data/user/0/{self.APP_ID}/databases/bluecoins.fydb" > {self.DB}.fydb',
+            shell=True,
+            check=True,
+        )
+
+    def cli_command_run(self, cli_command: str) -> None:
+        subprocess.run(f'poetry run bluecoins-cli {self.DB}.fydb {cli_command}', shell=True, check=True)
```

Comment:
Smart move! But don't rely on Poetry here. You have two options:

* Call `bluecoins-cli` as a script with `subprocess`
* `from bluecoint_cli import cli`, then call it directly